src/Ethereum/ABI/Numbers.h

```diff
@@ -15,132 +15,160 @@ static constexpr std::size_t encodedIntSize = 32;
 
 // uint256_t
 
-bool is_dynamic(uint256_t) {
+inline bool is_dynamic(uint256_t) {
     return false;
 }
 
-std::size_t size(uint256_t) {
+inline std::size_t size(uint256_t) {
     return encodedIntSize;
 }
 
-void encode(uint256_t value, Data& data) {
-    Data bytes = store(value);
+void encode(uint256_t value, Data& data);
 
-    append(data, Data(encodedIntSize - bytes.size()));
-    append(data, bytes);
-}
+bool decode(const Data& encoded, uint256_t& decoded, size_t& offset_inout);
 
-std::string type_string(uint256_t value) {
+inline std::string type_string(uint256_t value) {
     return "uint256";
 }
 
 // int256_t
 
-bool is_dynamic(int256_t) {
+inline bool is_dynamic(int256_t) {
     return false;
 }
 
-std::size_t size(int256_t) {
+inline std::size_t size(int256_t) {
     return encodedIntSize;
 }
 
-void encode(int256_t value, Data& data) {
+inline void encode(int256_t value, Data& data) {
     encode(static_cast<uint256_t>(value), data);
 }
 
-std::string type_string(int256_t value) {
+template<typename T>
+inline bool decodeGenericNumber(const Data& encoded, T& decoded, size_t& offset_inout) {
```

Comment:
maybe just `decodeNumber`? T is already a generic type

Reply:
OK, accepted, thx